ogorek.go

```diff
@@ -448,8 +458,19 @@ type Call struct {
 }
 
 func (d *Decoder) reduce() error {
-	args := d.pop().([]interface{})
-	class := d.pop().(Class)
+	if len(d.stack) < 2 {
```

Comment:
I initially started this way - the way you suggests but in for non-single pop the error handling adds imho too much. What is also important in some cases - see e.g. your next comment about loadAppend() there is not only pop, but also later `d.stack[len(d.stack) - 1]` for which we _have_ to add check anyway.

So this way we can have one check in the prologue covering whole  function for particular op handling.

I agree however that hiding error from pop is not good, so it wil be amended.